src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+                            new XElement(
+                                snippetNamespace + "SnippetTypes",
+                                new XElement(snippetNamespace + "SnippetType", new XText("Expansion"))),
+                            new XElement(snippetNamespace + "Title", new XText(methodName)),
+                            new XElement(snippetNamespace + "Author", "Microsoft"),
+                            new XElement(snippetNamespace + "Description"),
+                            new XElement(snippetNamespace + "HelpUrl"),
+                            new XElement(snippetNamespace + "Shortcut", methodName)),
+                        new XElement(
+                            snippetNamespace + "Snippet",
+                            new XElement(snippetNamespace + "Declarations", declarations.ToArray()),
+                            new XElement(
+                                snippetNamespace + "Code",
+                                new XAttribute(snippetNamespace + "Language", "csharp"),
+                                new XCData(template.ToString()))))));
+        }
+
+        private void OnModelUpdated(object sender, Model e)
+        {
+            AssertIsForeground();
+
+            if (e is null)
+            {
+                return;
+            }
+
+            if (!_state.IsFullMethodCallSnippet)
+            {
+                return;
+            }
+
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return;
+            }
+
+            var newSymbolKey = (e.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
+            var newSymbol = newSymbolKey.Resolve(document.Project.GetRequiredCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None).GetAnySymbol();
+            if (newSymbol is not IMethodSymbol method)
+                return;
+
+            MoveToSymbol(_state._symbols, method, CancellationToken.None);
+        }
+
+        private async Task<ImmutableArray<ISymbol>> GetSymbolsAsync(
+            SnapshotPoint caretPosition,
+            CancellationToken cancellationToken)
+        {
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return ImmutableArray<ISymbol>.Empty;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var token = await semanticModel.SyntaxTree.GetTouchingTokenAsync(caretPosition.Position, cancellationToken).ConfigureAwait(false);
+            var semanticInfo = semanticModel.GetSemanticInfo(token, document.Project.Solution.Workspace, cancellationToken);
+            return semanticInfo.ReferencedSymbols;
```

Comment:
so, given that we're interacting with sighelp, and we'll be moving to the partiular method it says the user is on, could we not grab the *set* of methods entirely from it once the sighelp model is computed and forwarded to us?  that seems cleaner as we're effectively not reimplementing anything but instead *always* just trusting what sighelp is telling us, no?

Reply:
It would break for signature help during nested calls. For example, consider a case where `WriteLine` is part of a Insert Full Method Call operation but `ToString` is not:

```
Console.WriteLine(x.ToString($$))
```

At the `$$`, signature help applies to `ToString`, and we don't return to `WriteLine`'s signature help until the `)` is typed.

Reply:
➡️ Leaving this for #50637